lib/rules/semi.js

```diff
@@ -53,7 +53,7 @@ module.exports = {
 
     create(context) {
 
-        const OPT_OUT_PATTERN = /[\[\(\/\+\-]/; // One of [(/+-
+        const OPT_OUT_PATTERN = /[\[\(\/](\+[^\+]|\-[^\-])/; // One of [(/+-, but not ++ or --
```

Comment:
Could you add tests like below?

    var a = b;
    + c

This semicolon cannot be removed.
But this fix seems to make the rule warning this semicolon.

---

I think this RegExp can be more simple.

    const OPT_OUT_PATTERN = /^[[(/+-]$/;